Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py

```diff
@@ -2203,10 +2218,22 @@ def parse_incident_from_item(item):     # pragma: no cover
                                     and header.name.lower() != "content-type"
                             ):
                                 try:
-                                    attached_email.add_header(header.name, header.value)
+                                    if header.name.lower() == "message-id":
+                                        # Handle a case where a Message-ID header was NOT already in attached_email,
+                                        # and instead is coming from attachment.item.headers.
+                                        # Meaning it wasn't handled in handle_attached_email_with_incorrect_message_id function
+                                        # and instead it is handled here using handle_incorrect_message_id function.
+                                        correct_message_id = handle_incorrect_message_id(header.value)
+                                        if (header.name.lower(), correct_message_id) not in attached_email_headers:
+                                            attached_email.add_header(header.name, correct_message_id)
+                                    else:
+                                        attached_email.add_header(header.name, header.value)
                                 except ValueError as err:
                                     if "There may be at most" not in str(err):
                                         raise err
+                                except IndexError as err:
+                                    demisto.debug('IndexError exception is raised from parse_incidents_from_item')
+                                    raise err
```

Comment:
why do we need this? Don't we get it with line numbers when an exception is raised?

Reply:
It was part of the removing, removing.